server/core/src/irods_signal.cpp

```diff
@@ -1,38 +1,87 @@
 #include "irods_signal.hpp"
-#include "irods_stacktrace.hpp"
-#include "irods_logger.hpp"
 
-#include <string.h>
-#include <signal.h>
+#include "irods_default_paths.hpp"
 
+#include <boost/stacktrace/safe_dump_to.hpp>
+
+#include <csignal>
+#include <cstdio>
 #include <cstdlib>
-#include <sstream>
+#include <cstring>
+#include <ctime>
+#include <system_error>
+#include <charconv>
 
-// Define signal handlers for irods
+#include <unistd.h>
 
 extern "C"
+void stacktrace_signal_handler(int _signal)
 {
-    /// @brief Signal handler for seg faults
-    static void segv_handler(int _signal)
-    {
-        using log = irods::experimental::log;
+    timespec ts;
+
+    if (clock_gettime(CLOCK_REALTIME, &ts) != 0) {
+        std::exit(_signal);
+    }
+
+    const auto max_path_length = 512;
```

Comment:
why 512?

Reply:
No particular reason. I figured this would be large enough to hold a path. Maybe this should match the largest path Linux supports.

Reply:
seems defensible.

Reply:
changed to use _POSIX_PATH_MAX which is guaranteed to be 256.

a buffer of this size is safe given that we're dealing with a relative path.